server.js

```diff
@@ -15,6 +18,133 @@ app.get("/", function (request, response) {
 
 // TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
+// Search bookings by date or by name in the email , firstName or surname
+
+app.get("/bookings/search", (req, res) => {
```

Comment:
Your logic here is okay, but what if someone searches `/bookings/search?term=jimi&date=2017-11-11`. One way you could resolve this is by breaking your functions up a bit more and then combining them. Maybe not perfect, but here's a possible complete controller here:

```
app.get("/bookings/search", (req, res) => {
  const { date, term } = req.query;
  const dateAsJsDate = new Date(date);

  const isDateInBookingRange = (booking) => {
    const bookingStart = new Date(booking.checkInDate);
    const bookingEnd = new Date(booking.checkOutDate);
    return dateAsJsDate >= bookingStart && dateAsJsDate <= bookingEnd;
  };

  const isTermInBooking = (booking) => {
    return [
      booking.firstName.toLowerCase(),
      booking.surname.toLowerCase(),
      booking.email.toLowerCase(),
    ].includes(term.toLowerCase());
  };

  const searchResult = bookings.filter((booking) => {
    return isDateInBookingRange(booking) && isTermInBooking(booking);
  });

  res.status(200).send(searchResult);
});
```